helm/templates/deployment.yaml

```diff
@@ -0,0 +1,80 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "sql-exporter.fullname" . }}
+  labels:
+    app: {{ include "sql-exporter.fullname" . }}
+    chart: {{ template "sql-exporter.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    {{- include "sql-exporter.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "sql-exporter.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        rollme: {{ randAlphaNum 5 | quote }}
+      labels:
+        app: {{ include "sql-exporter.fullname" . }}
+        chart: {{ template "sql-exporter.chart" . }}
+        release: {{ .Release.Name }}
+        {{- include "sql-exporter.selectorLabels" . | nindent 8 }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      volumes:
+        - name: sql-exporter
+          secret:
+            secretName: {{ include "sql-exporter.fullname" . }}
+        {{- if .Values.collectorFiles}}
+        - name: sql-collector
+          configMap:
+            name: {{ include "sql-exporter.fullname" . }}
+        {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args: ["-config.file=/etc/sql_exporter/sql_exporter.yml"]
+          volumeMounts:
+          - name: sql-exporter
+            readOnly: true
+            mountPath: /etc/sql_exporter/
+          {{- if .Values.collectorFiles}}
+          - name: sql-collector
+            readOnly: true
+            mountPath: /etc/sql_exporter/collectors/
+          {{- end }}
+          {{- if .Values.livenessProbe }}
+          livenessProbe: {{- toYaml .Values.livenessProbe | nindent 12 }}
```

Comment:
WDYT if we use `/healthz` for livenessProbe/readinessProbe of the container? I don't think users would put complicated logic behind `readinessProbe` (e.g. querying the actual backend from kubelet, this might be not the best idea), so we could just return HTTP 200 for both for starters.

In addition (just for context), there's a new feature coming, which would return prometheus output with `scrape_errors_total` counter instead of HTTP 500 (if target is named). Users then can configure this metric and check for scrape failures.

Reply:
> WDYT if we use /healthz for livenessProbe/readinessProbe 

To be honest, I've copies this part from a PR to the old SQL_exporter repo. So yeah, I'll update them

Reply:
> In addition (just for context), there's a new feature coming, which would return prometheus output with scrape_errors counter instead of HTTP 500 (if target is named). Users then can configure this metric and check for scrape failures.

That sounds nice, I can take care of tests once it's released